apps/api/v2/src/modules/organizations/services/organizations-schedules.service.ts

```diff
@@ -0,0 +1,24 @@
+import { SchedulesService_2024_06_11 } from "@/ee/schedules/schedules_2024_06_11/services/schedules.service";
+import { UsersRepository } from "@/modules/users/users.repository";
+import { Injectable } from "@nestjs/common";
+
+@Injectable()
+export class OrganizationsSchedulesService {
+  constructor(
+    private readonly schedulesService: SchedulesService_2024_06_11,
+    private readonly usersRepository: UsersRepository
+  ) {}
+
+  async getOrganizationSchedules(organizationId: number) {
+    const users = await this.usersRepository.getOrganizationUsers(organizationId);
+    const usersIds = users.map((user) => user.id);
+
+    const schedules = await Promise.all(
+      usersIds.map((userId) => {
+        return this.schedulesService.getUserSchedules(userId);
+      })
+    );
+
+    return schedules.flat();
+  }
```

Comment:
I'm worried this might overload our database very quickly if there are lots of users with lots of schedules

Reply:
Good point - to be safe got ride of Promise.all in [refactor: dont use Promise.all](https://github.com/calcom/cal.com/pull/15584/commits/09e7b35a7e8e33be5c8533235b26c050610145f3)